test/OverwhelmingSupportAutoDelegate.t.sol

```diff
@@ -336,3 +336,91 @@ contract OZGovernorBlockNumberModeSetSupportThreshold is OverwhelmingSupportAuto
+    rollOrWarpToTimepoint(_proposalDeadline - _blocksWithinVotingWindow);
+
+    // Call the function and ensure no revert
+    autoDelegate.checkVoteRequirements(address(governor), _proposalId);
+  }
+
+  function testFuzz_RevertIf_OutsideVotingWindow(
+    uint256 _proposalId,
+    uint256 _blocksOutsideVotingWindow,
+    uint256 _proposalDeadline
+  ) public {
+    _proposalDeadline = bound(_proposalDeadline, getTimepoint() + governor.votingPeriod(), type(uint48).max);
+    _blocksOutsideVotingWindow = bound(_blocksOutsideVotingWindow, votingWindow + 1, _proposalDeadline);
+
+    // Set the proposal details
+    governor.__setProposals(_proposalId, _proposalDeadline, /*For*/ 0, /*Against*/ 0);
+    rollOrWarpToTimepoint(_proposalDeadline - _blocksOutsideVotingWindow);
+
+    // Expect revert due to being outside the voting window
+    vm.expectRevert(OverwhelmingSupportAutoDelegate.OverwhelmingSupportAutoDelegate__OutsideVotingWindow.selector);
+    autoDelegate.checkVoteRequirements(address(governor), _proposalId);
+  }
+
+  function testFuzz_RevertIf_InsufficientForVotes(
+    uint256 _proposalId,
+    uint256 _blocksWithinBuffer,
+    uint256 _forVotesBIP
+  ) public {
+    _blocksWithinBuffer = bound(_blocksWithinBuffer, 0, votingWindow);
+    _forVotesBIP = bound(_forVotesBIP, 0, subQuorumBips - 1);
+    uint256 _forVotes = governor.quorumVotes() * _forVotesBIP / BIP;
+
+    // Set the proposal details
+    governor.__setProposals(_proposalId, proposalDeadline, _forVotes, /*Against*/ 0);
+    rollOrWarpToTimepoint(proposalDeadline - _blocksWithinBuffer);
+
+    // Expect revert due to insufficient FOR votes
+    vm.expectRevert(OverwhelmingSupportAutoDelegate.OverwhelmingSupportAutoDelegate__InsufficientForVotes.selector);
+    autoDelegate.checkVoteRequirements(address(governor), _proposalId);
+  }
+
+  function testFuzz_RevertIf_BelowSupportThreshold(
+    uint256 _proposalId,
+    uint256 _blocksWithinBuffer,
+    uint256 _totalVotes,
+    uint256 _forVotes,
+    uint256 _againstVotes
+  ) public {
+    _blocksWithinBuffer = bound(_blocksWithinBuffer, 0, votingWindow);
+    uint256 _subQuorumVotes = (governor.quorumVotes() * subQuorumBips / BIP);
+    _totalVotes = bound(_totalVotes, _subQuorumVotes * 2, GOV_SUPPLY);
+    _forVotes = bound(_forVotes, _subQuorumVotes, _totalVotes * supportThreshold / BIP - 1);
+    _againstVotes = _totalVotes - _forVotes;
+
+    // Set the proposal details
+    governor.__setProposals(_proposalId, proposalDeadline, _forVotes, _againstVotes);
+    rollOrWarpToTimepoint(proposalDeadline - _blocksWithinBuffer);
+
+    // Expect revert due to insufficient support threshold
+    vm.expectRevert(OverwhelmingSupportAutoDelegate.OverwhelmingSupportAutoDelegate__BelowSupportThreshold.selector);
+    autoDelegate.checkVoteRequirements(address(governor), _proposalId);
+  }
+}
+
+contract OZGovernorBlockModeCheckVoteRequirements is CheckVoteRequirementsTest {
```

Comment:
We have the abstract test, but only one concrete version. Can you explain why that is? Seems to me we should either have:

* Only one concrete test, no abstract version
* OR the abstract test should have two concrete inheritors for the different cases

Reply:
DONE: created to instances.. [c18a7ba](https://github.com/withtally/stGOV/pull/87/commits/c18a7ba00e7ef7b47d2fac09dd33f1c9e2a36b9f)